lib/assertions.js

```diff
@@ -700,7 +700,11 @@ module.exports = function (expect) {
         '<object> to be (a map|a hash|an object) whose values [exhaustively] satisfy <assertion>'
     ], function (expect, subject, nextArg) {
         expect.errorMode = 'nested';
-        expect(subject, 'not to equal', {});
+        if (expect.subjectType.is('array')) {
```

Comment:
It would be nicer if we had `not to be empty` that worked on both `array-like` and `object`.